File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDepth = function(root) {
if(root===null) return 0;
        let l = minDepth(root.left);
        let r = minDepth(root.right);
        if(root.left==null && root.right==null) return 1;
        if(root.left==null) return 1+r;
        if(root.right==null) return 1+l;
        return 1+Math.min(r,l);
};

File: 0111-minimum-depth-of-binary-tree/0111-minimum-depth-of-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int solve(TreeNode* root) {
        if(root->left==NULL && root->right==NULL) return 1;
        if(root->left==NULL) return 1+solve(root->right);
        if(root->right==NULL) return 1+solve(root->left);
        return 1+ min(solve(root->left), solve(root->right));
    }
    int minDepth(TreeNode* root) {
        if(root==NULL) return 0;
        return solve(root);
    }
};